pkg/routeagent_driver/cabledriver/xfrm_cleanup_handler.go

```diff
@@ -43,5 +44,12 @@ func (h *xrfmCleanup) GetNetworkPlugins() []string {
 func (h *xrfmCleanup) TransitionToNonGateway() error {
 	logger.Info("Transitioned to non-Gateway, cleaning up the IPsec xfrm rules")
 
-	return netlink.DeleteXfrmRules(k8snet.IPv4) //nolint:wrapcheck  // No need to wrap this error
+	errv6 := netlink.DeleteXfrmRules(k8snet.IPv6) //nolint:wrapcheck  // No need to wrap this error
+	errv4 := netlink.DeleteXfrmRules(k8snet.IPv4) //nolint:wrapcheck  // No need to wrap this error
+
+	if errv6 != nil || errv4 != nil {
+		return errors.New("error deleting xfrm rules")
+	}
+
+	return nil
```

Comment:
Let's aggregate the errors so we don't lose them.

`import utilerrors "k8s.io/apimachinery/pkg/util/errors"`

```suggestion
	return utilerrors.NewAggregate([]error{errv6, errv4})
```